File: bench/data-structures-vector/3.cpp
#include <iostream>
#include <vector>
#include <ctime>
#include <cstdlib>
#include <chrono>

using namespace std;

int main(int argc, char const *argv[])
{
	srand(time(NULL));

	vector<int> vec;

	int S = 100000000;
	int n = 100;
	if (argc >= 2)
	{
		S = atoi(argv[1]);

		if (argc == 3)
		{
			n = atoi(argv[2]);
		}
	}

	for (int i = 0; i < S; i++)
	{
		vec.push_back(42);
	}

	auto start = chrono::steady_clock::now();

	for (int i = 0; i < n; i++)
	{
		vec.erase(vec.begin() + (rand() % vec.size()));
	}

	cout << chrono::duration_cast<chrono::nanoseconds>(chrono::steady_clock::now() - start).count() << endl;

	return 0;
}

File: bench/data-structures-vector/3.go
package main

import (
	"math/rand"
	"os"
	"strconv"
	"fmt"
	"time"
)

func main() {
	rand.Seed(time.Now().UnixNano())

	var S = 100000000
	var n = 100
	var args = os.Args[1:]
	if len(args) >= 1 {
		S, _ = strconv.Atoi(args[0])
		if len(args) >= 2 {
			n, _ = strconv.Atoi(args[1])
		}
	}

	var arr = make([]int, S)

	for i := 0; i < S; i++ {
		arr[i] = 42
	}

	var start = time.Now()

	for i := 0; i < n; i++ {
		var j = rand.Int() % S
		arr = append(arr[:j], arr[j+1:]...)
	}

	fmt.Println(time.Now().Sub(start).Nanoseconds())
}
